src/java.desktop/share/classes/java/awt/Robot.java

```diff
@@ -214,11 +220,42 @@ private static void checkIsScreenDevice(GraphicsDevice device) {
      * while the subsequent mousePress and mouseRelease can be
      * delivered to the correct location
      *
-     * @param x         X position
-     * @param y         Y position
+     * @param x X position
+     * @param y Y position
      */
     public synchronized void mouseMove(int x, int y) {
-        peer.mouseMove(x, y);
+        int leastXDiff = Integer.MAX_VALUE;
+        int leastYDiff = Integer.MAX_VALUE;
+        int finX = x;
+        int finY = y;
+
+        for (Rectangle screenBounds : allScreenBounds) {
+            int closestX = Math.min(Math.max(x, screenBounds.x), screenBounds.x + screenBounds.width);
+            int closestY = Math.min(Math.max(y, screenBounds.y), screenBounds.y + screenBounds.height);
```

Comment:
Shouldn't the width and height have `- 1`? I mean if the screen resolution is 1920×1080, the highest valid coordinates are 1919 and 1079 for <var>x</var> and <var>y</var> correspondingly.